src/auth/local/local-auth.guard.spec.ts

```diff
@@ -0,0 +1,133 @@
+    const moduleRef = await Test.createTestingModule({
+      imports: [PassportModule],
+      providers: [
+        LocalAuthGuard,
+        {
+          provide: AuthService,
+          useValue: {
+            // 模擬AuthService中的方法
+            validateUser: jest.fn(),
+          },
+        },
+        LocalStrategy,
+      ],
+    }).compile();
+
+    localAuthGuard = moduleRef.get<LocalAuthGuard>(LocalAuthGuard);
+    authService = moduleRef.get<AuthService>(AuthService);
+  });
+
+  it("should be defined", () => {
+    expect(localAuthGuard).toBeDefined();
+  });
+
+  it("should return true if user is valid", async () => {
+    const mockUser = {
+      email: "test@example.com",
+      id: 1,
+    };
+
+    const mockRequest = {
+      body: {
+        account: "test",
+        password: "password",
+      },
+    };
+
+    const mockResponse = {};
+
+    const mockExecutionContext = {
+      switchToHttp: () => ({
+        getRequest: () => mockRequest,
+        getResponse: () => mockResponse,
+      }),
+    } as ExecutionContext;
+
+    jest
+      .spyOn(authService, "validateUser")
+      .mockImplementation(async () => mockUser);
+
+    const result = await localAuthGuard.canActivate(mockExecutionContext);
+
+    expect(result).toBe(true);
+  });
+
+  it("should return Forbidden and 403 http code when account information is wrong.", async () => {
+    const mockResponse = {};
+    const mockExecutionContext = {
+      switchToHttp: () => ({
+        getRequest: () => ({
+          body: {
+            account: "test",
+            password: "password",
+          },
+        }),
+        getResponse: () => mockResponse,
+      }),
+    } as ExecutionContext;
+
+    jest
+      .spyOn(authService, "validateUser")
+      .mockImplementation(async () => null);
+
+    try {
+      await localAuthGuard.canActivate(mockExecutionContext);
+    } catch (error) {
+      if (error instanceof HttpException) {
+        expect(error).toBeInstanceOf(ForbiddenException);
+        expect(error.getResponse()).toEqual({
+          message: ["Account or password is wrong."],
+          statusCode: 403,
+        });
+      }
+    }
+  });
+
+  it("should return BadRequest and 400 http code when field format validation failed.", async () => {
+    const mockResponse = {};
+    const mockExecutionContext = {
+      switchToHttp: () => ({
+        getRequest: () => ({
+          body: {
+            account: "test",
+          },
+        }),
+        getResponse: () => mockResponse,
+      }),
+    } as ExecutionContext;
+
+    try {
+      await localAuthGuard.canActivate(mockExecutionContext);
+    } catch (error) {
+      if (error instanceof HttpException) {
+        expect(error).toBeInstanceOf(BadRequestException);
+        expect(error.getResponse()).toEqual({
+          error: "Bad Request",
+          message: [
+            "password 必須長度大於等於8個字。",
+            "password 為必填欄位。",
+          ],
+          statusCode: 400,
+        });
+      }
```

Comment:
is the if clause required? why don't just directly assert it?

Reply:
When asserting in getResponse, it will consider the error as type "any".
So... need the if clause is required.

Reply:
OK.
[refactor: delete if clause based on suggestions](https://github.com/Cophr/backend/pull/10/commits/5b4b4fd7e32071b0e23643608103825233a0fba9)